integration/test_collection.py

```diff
@@ -1467,18 +1475,48 @@ class Data(TypedDict):
 
     collection.data.insert_many([DataObject(properties={"data": i}) for i in range(10)])
 
-    # make sure a new iterator resets the internal state
-    for _ in range(3):
-        # get the property and sort them - order returned by weaviate is not identical to the order inserted
-        all_data: list[int] = sorted(
-            [
-                int(obj.properties["data"])
-                for obj in collection.iterator(return_metadata=metadata, return_properties=Data)
-                if metadata is None or obj.metadata.creation_time_unix is not None
-            ]
-        )
+    iter_ = collection.iterator(
+        return_metadata=return_metadata, return_properties=return_properties
+    )
+
+    # Expect everything back
+    if return_metadata is None and return_properties is None:
+        all_data: list[int] = sorted([int(obj.properties["data"]) for obj in collection.iterator()])
+        assert all_data == list(range(10))
+
+        creation_time_unixes = [obj.metadata.creation_time_unix is not None for obj in iter_]
+        assert all(creation_time_unixes)
+        scores = [obj.metadata.score is not None for obj in iter_]
+        assert all(scores)
+
+    # Expect only metadata with only creation_time_unix
+    elif return_metadata is not None and return_properties is None:
+        all_props = [obj.properties == {} for obj in iter_]
+        assert all(all_props)
+
+        creation_time_unixes = [obj.metadata.creation_time_unix is not None for obj in iter_]
+        assert all(creation_time_unixes)
+        scores = [obj.metadata.score is None for obj in iter_]
+        assert all(scores)
```

Comment:
do you just want to do `all(obj.metadata.score is None for obj in iter_)`? (same for the others)

Reply:
That looks a lot nicer! Will change :D